axiom/nr_span_event.h

```diff
@@ -21,9 +21,30 @@ typedef struct _nr_span_event_t nr_span_event_t;
 typedef enum {
   NR_SPAN_GENERIC,
   NR_SPAN_HTTP,
-  NR_SPAN_DATASTORE
+  NR_SPAN_DATASTORE,
+  NR_SPAN_MESSAGE
 } nr_span_category_t;
 
+/*
+ * The spankinds a span may fall into.
+ * This is set according to:
+ * 1) guidelines in agent-specs which state datastore and http spans set
+ * span.kind to client and further states that generic span.kind is unset
+ *
+ * 2) for message spans follow guidance here:
+ * https://opentelemetry.io/docs/specs/semconv/messaging/messaging-spans/
+ * which states that span.kind is
+ * a) producer when the operation type is create or send(if the context is
+ * create) b) client when the operation type is create or send(if the context is
+ * NOT create) c) consumer when the operation type is process
+ */
+typedef enum {
+  NR_SPAN_PRODUCER,
+  NR_SPAN_CLIENT,
+  NR_SPAN_CONSUMER,
+  NR_SPAN_NO_SPANKIND
```

Comment:
With this naming convention it's hard to tell if `NR_SPAN_PRODUCER` is span category or span kind. Consider using `NR_SPAN_KIND_` prefix.

Reply:
Please see: https://github.com/newrelic/newrelic-php-agent/pull/991/commits/862604b081029206c2072275588076691d30560a